pkg/issuer/acme/issue.go

```diff
@@ -48,21 +50,23 @@ func (a *Acme) obtainCertificate(ctx context.Context, crt *v1alpha1.Certificate)
 		return nil, nil, fmt.Errorf("error getting order details: %v", err)
 	}
 
-	if order.Status != acmeapi.StatusReady {
-		err := fmt.Errorf("expected certificate status to be %q, but it is %q", acmeapi.StatusReady, order.Status)
-		// print a more helpful message to users when an order is marked 'valid'.
-		// this happens when all challenges have been completed successfully, but
-		// the acme server has not finished processing the order.
```

Comment:
This comment was incorrect. 'ready' indicates an order is ready to be finalized, whereas 'valid' indicates the order has already been finalized and can be retrieved with GetCertificate.